### PR TITLE
Create custom tab bar

### PR DESCRIPTION
**Pull request by dennisfedorko**
Please review to make sure everything makes sense

Obviously we can change things around, waiting on that floating player view to be implemented to add that in, but this is what I came up with, its by no means perfect, but allows us to do some custom behavior/animations that we couldn't do with the vanilla UITabBarController.

### CODE REVIEW

File: Podcast/TabBarController.swift [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            
+            let blurEffect = UIBlurEffect(style: UIBlurEffectStyle.extraLight)
+            let blurEffectView = UIVisualEffectView(effect: blurEffect)
+
+            blurEffectView.frame = tabBarContainerView.bounds
+            blurEffectView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+            
+            tabBarContainerView.addSubview(blurEffectView)
+            
+        } else {
```
Comment (doctor3w) [inline Podcast/TabBarController.swift]: Do you need this else if you set it above before the if-statement?

File: Podcast/TabBarController.swift [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            tabBarButtons.append(newTabBarButton)
+            
+            xOffset += tabBarButtonWidth
+        }
+        
+    }
+    
+    func didPressTabBarButton(tabBarButton: UIButton) {
+        
+        guard let tabBarButtonIndex = tabBarButtons.index(of: tabBarButton) else { return }
```
Comment (doctor3w) [inline Podcast/TabBarController.swift:94]: Why not use button tag as index rather than search through an array?

**Review by doctor3w — APPROVED** [review]

File: Podcast/TabBarController.swift (hunk 1 above)
Comment (dennisfedorko) [reply Podcast/TabBarController.swift]: agreed

File: Podcast/TabBarController.swift (hunk 2 above)
Comment (dennisfedorko) [reply Podcast/TabBarController.swift:94]: im not a fan of using tags, its a 4 button array, so search is ok

Comment (doctor3w) [reply Podcast/TabBarController.swift:94]: Alright sounds good

**Pull request merged**